internal/kinds/capp/resourcemanagers/ksvc.go

```diff
@@ -0,0 +1,189 @@
+					ClaimName: nfsVolume.Name,
+				},
+			},
+		})
+	}
+	return volumes
+}
+
+// CleanUp attempts to delete the associated KnativeService for a given Capp resource.
+func (k KnativeServiceManager) CleanUp(capp cappv1alpha1.Capp) error {
+	resourceManager := rclient.ResourceManagerClient{Ctx: k.Ctx, K8sclient: k.K8sclient, Log: k.Log}
+	ksvc := knativev1.Service{}
+	if err := resourceManager.DeleteResource(&ksvc, capp.Name, capp.Namespace); err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to delete KnativeService %q: %w", ksvc.Name, err)
+	}
+	return nil
+}
+
+// IsRequired determines if a Knative service (ksvc) is required based on the Capp's spec.
+func (k KnativeServiceManager) IsRequired(capp cappv1alpha1.Capp) bool {
+	return capp.Spec.State == cappEnabledState
+}
+
+// isResumed checks whether the state changed from disabled to enabled.
+func (k KnativeServiceManager) isResumed(capp cappv1alpha1.Capp) bool {
+	return capp.Status.StateStatus.State == cappDisabledState && k.IsRequired(capp) &&
+		!capp.Status.StateStatus.LastChange.IsZero()
+}
+
+// Manage creates or updates a KnativeService resource based on the provided Capp if it's required.
+// If it's not, then it cleans up the resource if it exists.
+func (k KnativeServiceManager) Manage(capp cappv1alpha1.Capp) error {
+	if k.IsRequired(capp) {
+		return k.createOrUpdate(capp)
+	}
+
+	// if the KSVC is not required then it means that the Capp is in disabled state,
+	// therefore it's needed to clean up existing KSVC instances
+	k.Log.Info("Disabling Capp")
+	k.EventRecorder.Event(&capp, corev1.EventTypeNormal, eventCappDisabled, fmt.Sprintf("Capp %s state changed to disabled", capp.Name))
+	return k.CleanUp(capp)
+}
+
+// createOrUpdate creates or updates a KSVC resource.
+func (k KnativeServiceManager) createOrUpdate(capp cappv1alpha1.Capp) error {
+	knativeServiceFromCapp := k.prepareResource(capp, k.Ctx)
+	knativeService := knativev1.Service{}
+	resourceManager := rclient.ResourceManagerClient{Ctx: k.Ctx, K8sclient: k.K8sclient, Log: k.Log}
+
+	if err := k.K8sclient.Get(k.Ctx, types.NamespacedName{Namespace: capp.Namespace, Name: capp.Name}, &knativeService); err != nil {
+		if errors.IsNotFound(err) {
+			if err := k.createKSVC(&capp, &knativeServiceFromCapp, resourceManager); err != nil {
+				return fmt.Errorf("failed to create KnativeService %q: %w", knativeServiceFromCapp.Name, err)
+			}
+
+			if k.isResumed(capp) {
+				k.Log.Info("Capp resumed to enabled state")
+				k.EventRecorder.Event(&capp, corev1.EventTypeNormal, eventCappEnabled, fmt.Sprintf("Capp %q state changed to enabled", capp.Name))
+			}
+		} else {
+			return fmt.Errorf("failed to get KnativeService %q: %w", knativeService.Name, err)
+		}
+	} else if err := k.updateKSVCIfNeeded(&knativeService, &knativeServiceFromCapp, resourceManager); err != nil {
+		return fmt.Errorf("failed to update KnativeService %q: %w", knativeService.Name, err)
```

Comment:
if you doint it to every resource just pass the kind to the  resource client and the error parsing will be there